src/modules/75-cf/pages/onboarding/views/TestYourFlagView.tsx

```diff
@@ -0,0 +1,157 @@
+import { useParams } from 'react-router-dom'
+import { Color } from '@harness/design-system'
+import { Classes, Switch } from '@blueprintjs/core'
+import { String, useStrings } from 'framework/strings'
+import type { PlatformEntry } from '@cf/components/LanguageSelection/LanguageSelection'
+import routes from '@common/RouteDefinitions'
+import { ApiKey, Feature, useGetAllFeatures } from 'services/cf'
+import { useToggleFeatureFlag } from '@cf/hooks/useToggleFeatureFlag'
+import { TestFlagInfoView } from './TestFlagInfoView'
+import css from './TestYourFlagView.module.scss'
+
+const POLLING_INTERVAL_IN_MS = 3000
+
+export interface TestYourFlagViewProps {
+  flagInfo: Feature
+  language: PlatformEntry
+  apiKey: ApiKey
+  environmentIdentifier: string | undefined
+  testDone: boolean
+  setTestDone: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+export const TestYourFlagView: React.FC<TestYourFlagViewProps> = props => {
+  const { flagInfo, environmentIdentifier } = props
+  const { projectIdentifier, orgIdentifier, accountId: accountIdentifier } = useParams<Record<string, string>>()
+  const { getString } = useStrings()
+  const toggleFeatureFlag = useToggleFeatureFlag({
+    accountIdentifier,
+    orgIdentifier,
+    projectIdentifier,
+    environmentIdentifier: environmentIdentifier as string
+  })
+  const queryParams = useMemo(
+    () => ({
+      accountIdentifier,
+      orgIdentifier,
+      projectIdentifier,
+      environmentIdentifier,
+      identifier: flagInfo.identifier,
+      metrics: true
+    }),
+    [projectIdentifier, environmentIdentifier, accountIdentifier, orgIdentifier, flagInfo.identifier]
+  )
+  const { data, loading, refetch } = useGetAllFeatures({
+    lazy: true,
+    queryParams
+  })
+  const [checked, setChecked] = useState(false)
+  const timeoutIdRef = useRef<number>()
+
+  let link = routes.toCFFeatureFlagsDetail({
+    orgIdentifier: orgIdentifier as string,
+    projectIdentifier: projectIdentifier as string,
+    featureFlagIdentifier: flagInfo.identifier,
+    accountId: accountIdentifier
+  })
+  link = location.hash.startsWith('#/account/') ? '/#' + link : link
+
+  useEffect(() => {
+    if (!props.testDone) {
+      const pollingFn = (): void => {
+        if (!loading) {
+          refetch().finally(() => {
+            clearTimeout(timeoutIdRef.current)
+            timeoutIdRef.current = window.setTimeout(pollingFn, POLLING_INTERVAL_IN_MS)
+          })
+        } else {
+          clearTimeout(timeoutIdRef.current)
+          timeoutIdRef.current = window.setTimeout(pollingFn, POLLING_INTERVAL_IN_MS)
+        }
+      }
+      clearTimeout(timeoutIdRef.current)
+      timeoutIdRef.current = window.setTimeout(pollingFn, POLLING_INTERVAL_IN_MS)
+
+      return () => clearTimeout(timeoutIdRef.current)
+    }
+  }, [loading, refetch, props.testDone])
+
+  useEffect(() => {
+    if (!props.testDone && data) {
+      if (data.features?.[0].status?.status === 'active') {
```

Comment:
Could these be combined?
```suggestion
    if (!props.testDone && data?.features?.[0].status?.status === 'active') {
```

Reply:
This is existing code that I didn't write, I only moved it. Changing this is out of scope for this ticket.